src/Components/Web.Extensions/src/InputFile/RemoteBrowserFileStream.cs

```diff
@@ -0,0 +1,153 @@
+                        cancellationToken,
+                        _inputFileElement,
+                        File.Id,
+                        offset,
+                        chunkSize);
+
+                    if (!Convert.TryFromBase64String(base64, pipeBuffer.Span, out var bytesWritten))
+                    {
+                        throw new FormatException("A chunk with an invalid format was received.");
+                    }
+
+                    if (bytesWritten != chunkSize)
+                    {
+                        throw new InvalidOperationException(
+                            $"A chunk with size {bytesWritten} bytes was received, but {chunkSize} bytes were expected.");
+                    }
+
+                    writer.Advance(chunkSize);
+                    offset += chunkSize;
+                }
+                catch (Exception e)
+                {
+                    await writer.CompleteAsync(e);
+                    throw;
+                }
+
+                var result = await writer.FlushAsync();
+
+                if (result.IsCompleted)
+                {
+                    break;
+                }
+            }
+
+            await writer.CompleteAsync();
+        }
+
+        protected override async ValueTask<int> CopyFileDataIntoBuffer(long sourceOffset, Memory<byte> destination, CancellationToken cancellationToken)
+        {
+            if (_isReadingCompleted)
+            {
+                return 0;
+            }
+
+            int totalBytesCopied = 0;
+
+            while (destination.Length > 0)
+            {
+                var result = await _pipeReader.ReadAsync(cancellationToken);
+
+                if (result.IsCanceled)
+                {
+                    _isReadingCompleted = true;
+
+                    var exception = new OperationCanceledException(cancellationToken);
+                    await _pipeReader.CompleteAsync(exception);
+
+                    throw exception;
+                }
+
+                var bytesToCopy = (int)Math.Min(result.Buffer.Length, destination.Length);
+                var slice = result.Buffer.Slice(0, bytesToCopy);
+
+                slice.CopyTo(destination.Span);
+                _pipeReader.AdvanceTo(slice.End);
+
+                totalBytesCopied += bytesToCopy;
+
+                destination = destination.Slice(bytesToCopy);
+
+                if (result.IsCompleted && slice.Length == 0)
+                {
+                    _isReadingCompleted = true;
+
+                    await _pipeReader.CompleteAsync();
+
+                    break;
+                }
+            }
+
+            return totalBytesCopied;
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (_isDisposed)
+            {
+                return;
+            }
+
+            _fillBufferCts.Cancel();
+
+            if (disposing)
+            {
+                _fillBufferCts.Dispose();
```

Comment:
Don't dispose a CTS unless you're sure nothing is till using the token. In this case, it looks like FillBuffer could still be using the token given FillBuffer's completion is not observed/awaited. Given how it's used and that it's always canceled in the end, it'd be better not to dispose the CTS at all. It's not like there will be registrations hanging around indefinitely.